src/main/kotlin/org/stellar/walletsdk/Wallet.kt

```diff
@@ -111,4 +112,13 @@ class Wallet(
   fun removeAccountSigner(sourceAddress: String, signerAddress: String): Transaction {
     return addAccountSigner(sourceAddress, signerAddress, 0)
   }
+
+  // Submit transaction
+  fun submitTransaction(signedTransaction: Transaction): SubmitTransactionResponse {
+    try {
+      return server.submitTransaction(signedTransaction)
+    } catch (e: Error) {
+      throw Error("Transaction failed: $e")
+    }
+  }
```

Comment:
I don't think this function is useful as-is, but I think we can make it useful by providing some default error handling.

For example, we could add retry logic that resubmits the transaction when a `504` or `insufficient_fee` is received.

Essentially, we could build it to provide two possible states: success (included in a ledger) or failure (transaction expired or transaction invalid).

Reply:
We could retry when the response is `504` (timeout), but I'm not sure we should if the fee is too low. What if the transaction has a much lower fee than recommended? Or is it our recommendation for wallets to retry in both cases?

Reply:
> I'm not sure we should if the fee is too low. What if the transaction has a much lower fee than recommended?

If we don't retry in this case, then we'd need to throw an exception and let the developer decide what to do.

They have a couple options,

1. Call `submitTransaction()` again with the same transaction
2. Create a new transaction with a higher base fee and call `submitTransaction()` with the new transaction
3. Wrap the transaction in a `FeeBumpTransaction` and call `submitTransaction()` with the fee bump transaction
4. Error out in their app and inform the user that their transaction failed

I'm ok throwing an exception and letting them decide what to do, but we need to make sure we document how to handle the exception. We also should have helper functions that can fetch a base fee to be used from Horizon's `GET /fee_stats`.